tensorboard/data/experimental/experiment_from_dev.py

```diff
@@ -0,0 +1,162 @@
+        runs = []
+        tags = []
+        steps = []
+        wall_times = []
+        values = []
+        for response in stream:
+            # TODO(cais, wchargin): Display progress bar.
+            metadata = base64.b64encode(
+                response.tag_metadata.SerializeToString()
+            ).decode("ascii")
+            num_values = len(response.points.values)
+            runs.extend([response.run_name] * num_values)
+            tags.extend([response.tag_name] * num_values)
+            steps.extend(list(response.points.steps))
+            wall_times.extend(
+                [t.ToNanoseconds() / 1e9 for t in response.points.wall_times]
+            )
+            values.extend(list(response.points.values))
+
+        dataframe = pandas.DataFrame(
+            {
+                "run": runs,
+                "tag": tags,
+                "step": steps,
+                "wall_time": wall_times,
+                "value": values,
+            }
+        )
+        if pivot:
+            dataframe = dataframe.pivot_table(
+                ["value", "wall_time"], ["run", "step"], "tag",
+            )
+            if np.any(dataframe.isnull().values):
+                sys.stderr.write(
+                    "WARNING: pivoted DataFrame contains missing value(s). "
+                    "This is likely due to tags that contain uneven numbers of steps. "
+                    "Consider calling `get_scalar()` with `pivot=False`"
+                )
+                sys.stderr.flush()
+        return dataframe
+
+
+def get_api_client(api_endpoint=None):
+    server_info = _get_server_info(api_endpoint=api_endpoint)
+    channel_creds = grpc.ssl_channel_credentials()
+    credentials = auth.CredentialsStore().read_credentials()
+    if credentials:
+        channel_creds = grpc.composite_channel_credentials(
+            channel_creds, auth.id_token_call_credentials(credentials)
+        )
+    channel = grpc.secure_channel(
+        server_info.api_server.endpoint, channel_creds
+    )
+    return export_service_pb2_grpc.TensorBoardExporterServiceStub(channel)
+
+
+def _get_server_info(api_endpoint=None):
+    # TODO(cais): Add more plugins to the list when more plugin/data types
+    # are supported
+    plugins = ["scalars"]
+    if api_endpoint:
+        return server_info_lib.create_server_info(
+            DEFAULT_ORIGIN, api_endpoint, plugins
+        )
+    return server_info_lib.fetch_server_info(DEFAULT_ORIGIN, plugins)
+
+
+def _handle_server_info(info):
+    compat = info.compatibility
+    if compat.verdict == server_info_pb2.VERDICT_WARN:
+        sys.stderr.write("Warning [from server]: %s\n" % compat.details)
+        sys.stderr.flush()
+    elif compat.verdict == server_info_pb2.VERDICT_ERROR:
+        raise ValueError("Error [from server]: %s" % compat.details)
+    else:
+        # OK or unknown; assume OK.
+        if compat.details:
+            sys.stderr.write("%s\n" % compat.details)
+            sys.stderr.flush()
```

Comment:
This is never used; we probably want to nicely fail the initialization
on an error verdict, no?